### ISSUE TITLE
docker login fails in v4.11.0 for gitlab

### ISSUE THREAD

**Issue by flowcietytim** [docker/for-win#12871]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Actual behavior
`docker login registry.gitlab.com` fails in docker Desktop 4.11.0 wit error message
```
Error response from daemon: "docker login" is unsupported when Registry Access Management is blocking access to Docker Hub.
```

### Expected behavior
login succeeds and is confirmed

### Information
This might be related to the "minor change" in 4.11.0:
"Improved an error message from docker login if Registry Access Management is blocking the Docker engine’s access to Docker Hub."

The problem appeared after a clean install of Windows (10.0.19044 Build 19044) and Docker Desktop.
After uninstalling v4.11.0 and installing 4.10.1, login succeeds without any other changes.

This is not a Docker Business installation, so I'd expect to have nothing to do with Registry Access Management.

### Steps to reproduce the behavior

  1. Installs Docker 4.11.0 on Windows 10
  2. `docker login registry.gitlab.com`
  3. enter username and password/access token
  4. error occurs

**Comment by vijaytdh** [comment]
We are seeing the same issue. I noticed the release notes for 4.11.0 says:

```
Improved an error message from docker login if Registry Access Management is blocking the Docker engine’s access to Docker Hub.
```

Not sure if there is some bug in the implementation of this - because we definitely do not have this feature enabled in our org.

**Comment by tkintscher** [comment]
I can confirm and add that this seems to happen only
- with the Windows version
- after the upgrade to v4.11
- and even with previously already stored credentials.

**Comment by benediktblank** [comment]
Same issue here. Trying to login to Azure Registry but it fails with this error.

**Comment by pcomble** [comment]
I have the same problem with Docker Desktop 4.11.0 and Windows 10

**Comment by djs55** [comment]
Thanks for the report. It doesn't reproduce for me -- I can `docker login registry.gitlab.com` with Registry Access Management disabled. I suspect there is a bug when Registry Access Management is enabled, but I can't explain why it would activate if the feature isn't enabled in your org.

Could you
- reproduce the failure and upload a diagnostic and quote the ID here? 
- tell me if you have `C:\ProgramData\DockerDesktop\registry.json` or `C:\ProgramData\DockerDesktop\access.json` files, and if so, what they contain?
- tell me if you have access to Docker Hub, or if it is blocked by your firewall?

**Comment by dano2004-doc** [comment]
I have the same problem with Docker Desktop 4.11.0 and Windows 10, using a personal edition of docker desktop.

**Comment by mmathias01** [comment]
I am seeing the same issue. Cannot login to ECR and while I have a paid docker hub account I am not an organization or business and cannot turn on/off Registry Access Management.

**Comment by mmathias01** [comment]
In case it helps I have uploaded a diagnostic: 1B717F2-C406-4A6A-91A9-22822AC7CDF9/20220729184049 

For reference before I did this I:
- Turned it off and back on again
- Un-installed Docker Desktop
- Turned it off and back on again
- Cleaned up, moved, or removed all configuration files and folders both on the Windows 11 host, and the WSL2 VM
- Reinstalled Docker Desktop from a fresh download
- Crossed my fingers

TL;DR. This diagnostic should be on a clean, fresh install of Docker Desktop 4.11.0 on Windows 11 with WSL2

**Comment by pavelmaca** [comment]
I just come across this error on fresh new installation of W10 and Docker Desktop.

- Windows 10 (build 19042.1706)
- Docker Desktop 4.11.0 (83626)
- WSL2 with Debian

Diagnostic ID: 5BBE4157-2463-4E59-9DB3-F38B1DC86E7A/20220730051743



```
docker login registry.gitlab.com
Username: xxx@xxx.xx
Password: 
Error response from daemon: "docker login" is unsupported when Registry Access Management is blocking access to Docker Hub.

The CLI "docker login" will not work in this configuration.

**
** You must sign in via the Docker Desktop UI instead.
**
```

**Comment by misaon** [comment]
Confirm bug +1

**Comment by sanctitygeorge** [comment]
I am also having the same issue here on Windows10 Desktop version 4.11.0

Error message:

Error response from daemon: "docker login" is unsupported when Registry Access Management is blocking access to Docker Hub.

The CLI "docker login" will not work in this configuration.

**
** You must sign in via the Docker Desktop UI instead.
**

**Comment by Pawior** [comment]
Have same bug

**Comment by PatriotRossii** [comment]
Confirm.

**Comment by wood-dev** [comment]
Having same bug here

**Comment by shaileshs-waters** [comment]
Have same bug.

**Comment by gblazevictq** [comment]
Same bug occurs on my computer:

- Windows 10,
- Docker 4.11.0,
- WSL 2 with Ubuntu 20.04.4 LTS,
- Docker personal plan,
- Used command ` az acr login --name {REDACTED}`

```
Error response from daemon: "docker login" is unsupported when Registry Access Management is blocking access to Docker Hub.
The CLI "docker login" will not work in this configuration.
**
** You must sign in via the Docker Desktop UI instead.
**
Login failed.
```

**Comment by cheynewallace** [comment]
Confirmed this is also happening trying to login to ECR.

I was able to fix one machine by logging in using the Docker desktop software but you can only login to one account at once, which is not useful when you use ECR, Dockerhub and Digital Ocean docker repos.

Our deployment scripts are broken due to this error

**Comment by jruhe-adesso** [comment]
+1

**Comment by isc-gsaito** [comment]
It also happens to me. I had to re-install Docker Desktop recently and it now presents me the same error.

**Comment by wilstdu** [comment]
Having the same error with fresh install of Windows 10 and Docker Desktop 4.11.0 (83626)

**Comment by djs55** [comment]
I've got a test build which hopefully fixes the issue. If you'd like to try it: https://desktop-stage.docker.com/win/main/amd64/83852/Docker%20Desktop%20Installer.exe

However I still don't understand the root cause fully, so I've added lots more debug logging to that build. If it still doesn't work, could you upload a diagnostics ID and paste it here? Thanks!

**Comment by wilstdu** [comment]
@djs55, I've tested your test build and the ECR login passed without any problems.

**Comment by dahovey** [comment]
@djs55 Fix worked for me.

**Comment by corleyscotte** [comment]
@djs55 Build and push worked for me using ECR login!

- Windows 10
- WSL2 Debian 11
- Issue present with Docker Desktop 4.11.0
- Issue resolved with 4.12.0

**Comment by cheynewallace** [comment]
@djs55  that build works for me too.. Iv switched over to that now.

**Comment by kurtu5** [comment]
@djs55 The v4.12.0 build didn't quite work for me.

I am a new user, using Win10 WSL2 CLI, and am going through the tutorial where I have to login to add a tag and had the issue.  So I downloaded your test build and ran it while the current desktop was running.  No special cleaning was done and I am not sure if that was needed.  It failed.  Well, I tried to reboot and it still failed thusly;

```
kurt@BlackRock:~/getting-started.devdir$ docker login -u kurtu5
Password: 
Error response from daemon: Get "http://ipc/registry/access": dial unix /mnt/host/wsl/docker-desktop/shared-sockets/host-services/backend.sock: connect: no such file or directory
```

wsl is under /mnt/wsl and not /mnt/host/wsl, so I hacked in this symlink temporarily and it worked

`root@BlackRock:/mnt# mkdir host ; ln -s /mnt/wsl/ host/wsl`

Lastly I run diag.  Hopefully this is the right way and it helps.

```
kurt@BlackRock:~/getting-started.devdir$ /mnt/c/Program\ Files/Docker/Docker/resources/com.docker.diagnose.exe gather -upload
  ...
Diagnostics ID:     2B2B73F9-242B-485F-89F4-A8DEDE7DBB99/20220802114350 (uploaded)
```

**Comment by xiandan-erizo** [comment]
confirm 插眼

**Comment by elad-levy** [comment]
Having the same bug trying to login to ECR

**Comment by koenhendriks** [comment]
Same bug here with windows 11 and docker 4.11.0

**Comment by larsskj** [comment]
+1

**Comment by amitic** [comment]
+1
windows 10/11 + docker 4.11.0

**Comment by stephen-turner** [comment]
Hi all, we don't need any more +1's. It would be helpful if you could try the patch release posted above and let us know if that fixes it (or especially if it doesn't).

**Comment by absta** [comment]
Thanks for the fix, it worked for me (I was pushing logging into ECR)

**Comment by cesarmesones** [comment]
Fix also worked for me, Windows 10.0.19044 Build 19044 + docker 4.11.0 + wsl2 ubuntu20.04

**Comment by benbob-sci** [comment]
I had the same issue with Windows 10, Docker Desktop 4.11.0... the v4.12.0 fix worked for me.

**Comment by Pixman** [comment]
> I had the same issue with Windows 10, Docker Desktop 4.11.0... the v4.12.0 fix worked for me.

Hello, I have same issue on Windows 11 Docker Desktop 4.11.0 (83626). Where did you download the version 4.12.0 ?
Thanks

**Comment by larsskj** [comment]
The above fix only works partly for me. I still cannot login using WSL2/Ubuntu, but the error is different this time:

```
Authenticating with existing credentials...
Login did not succeed, error: Error response from daemon: Get "http://ipc/registry/access": dial unix /mnt/host/wsl/docker-desktop/shared-sockets/host-services/backend.sock: connect: no such file or directory
Login with your Docker ID to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com to create one.
```

No matter what I do, `docker login` refuses to find the socket. However, for commands like `docker ps` it works and it can apparently find the socket.

`docker login` now works from the Windows CLI.

Windows 10 Enterprise/1909

**Comment by Pixman** [comment]
Hello, the 4.12.0 (83852) version work for me. The login time is extremely slow, but it's work fine.

**Comment by flowcietytim** [comment]
I just tested the login with the published fix and it works for me.

```
docker login registry.gitlab.com
Authenticating with existing credentials...
Login Succeeded
```
I
- was previously logged in with 4.10.1
- `docker logout`
- updated using the provided installer
- rebooted
- log in succeeded

> I've got a test build which hopefully fixes the issue. If you'd like to try it: https://desktop-stage.docker.com/win/main/amd64/83852/Docker%20Desktop%20Installer.exe
> 
> However I still don't understand the root cause fully, so I've added lots more debug logging to that build. If it still doesn't work, could you upload a diagnostics ID and paste it here? Thanks!

**Comment by nlessmann** [comment]
Upgrading to 4.11.1 resolved the issue for me

**Comment by jsnedeco** [comment]
4.12.0 worked for me.
Edition	Windows 10 Pro for Workstations
Version	21H2
Installed on	‎7/‎29/‎2022
OS build	19044.1826
Experience	Windows Feature Experience Pack 120.2212.4180.0

Using WSL2

**Comment by nicks** [comment]
This is fixed in v4.11.1 https://docs.docker.com/desktop/release-notes/#docker-desktop-4111
